src/common/src/config.rs

```diff
@@ -2420,19 +2424,56 @@ pub struct StorageMemoryConfig {
     pub meta_cache_capacity_mb: usize,
     pub meta_cache_shard_num: usize,
     pub shared_buffer_capacity_mb: usize,
-    pub compactor_memory_limit_mb: usize,
     pub prefetch_buffer_capacity_mb: usize,
     pub block_cache_eviction_config: EvictionConfig,
     pub meta_cache_eviction_config: EvictionConfig,
-    pub block_file_cache_flush_buffer_threshold_mb: usize,
+    pub data_file_cache_flush_buffer_threshold_mb: usize,
     pub meta_file_cache_flush_buffer_threshold_mb: usize,
 }
 
+pub struct CompactorMemoryConfig {
+    pub compactor_meta_cache_capacity_mb: usize,
+    pub compactor_memory_limit_mb: usize,
+}
+
 pub const MAX_META_CACHE_SHARD_BITS: usize = 4;
 pub const MIN_BUFFER_SIZE_PER_SHARD: usize = 256;
 pub const MAX_BLOCK_CACHE_SHARD_BITS: usize = 6; // It means that there will be 64 shards lru-cache to avoid lock conflict.
 
-pub fn extract_storage_memory_config(s: &RwConfig) -> StorageMemoryConfig {
+pub const COMPACTOR_MEMORY_PROPORTION: f64 = 0.1;
```

Comment:
Can we file an issue? I suppose more broadly across difference services, their memory ratios should sum up to `1.0`.